### ISSUE TITLE
Authenticator Social Sign In buttons causing RenderFlex overflow on smaller devices

### ISSUE THREAD

**Issue by jochemkleine** [aws-amplify/amplify-flutter#3689]
### Description

Following the tutorial found at https://ui.docs.amplify.aws/flutter/connected-components/authenticator/customization, I am creating an Authentication flow for my Flutter based mobile app (iOS + Android). 

The approach works well on most devices, but on smaller devices, such as the Pixel 5, implementing social sign in buttons (tried with Google and Apple) will cause a RenderFlex overflow. 

I've tried disabling the social login option and can confirm the rest of the SignInForm() implementation do not cause this error; they are correctly resized by changing the maxWidth BoxConstraint as shown in the tutorial. 

**The SignInForm implementation works as expected on:** 
Pixel 6 Emulator
iPhone Pro 14 Max Emulator

**And causes an error on:** 
Physical Pixel 5

### Categories

- [ ] Analytics
- [ ] API (REST)
- [ ] API (GraphQL)
- [ ] Auth
- [X] Authenticator
- [ ] DataStore
- [ ] Notifications (Push)
- [ ] Storage

### Steps to Reproduce

Please find the relevant code below. Deploying this on a **physical Pixel 5** will result in a rendering overflow error. 

**Relevant Code** 

Please find the CustomScaffold and related code below. 
````
class CustomScaffold extends StatelessWidget {
  final AuthenticatorState state;
  final Widget body;
  final Widget? footer;

  const CustomScaffold({
    required this.state,
    required this.body,
    required this.footer,
  });

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Padding(
        padding: const EdgeInsets.all(16),
        child: SingleChildScrollView(
          child: Column(
            children: [
              Padding(
                padding: EdgeInsets.only(top: 32),  // Similar padding to the example
                child: Center(
                  child: Text(
                    "App",
                    style: TextStyle(
                        fontSize: 36,
                        fontWeight: FontWeight.bold,
                        color: Colors.black
                    ), // Changed color to black to match example
                  ),
                ),
              ),
              Container(
                constraints: const BoxConstraints(maxWidth: 400),
                child: body,
              ),
            ],
          ),
        ),
      ),
      persistentFooterButtons: footer != null ? [footer!] : null,
    );
  }
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Authenticator(
      authenticatorBuilder: (BuildContext context, AuthenticatorState state) {
        switch (state.currentStep) {
          case AuthenticatorStep.signIn:
            return CustomScaffold(
              state: state,
              body: SignInForm(),
              footer: Row(
                mainAxisAlignment: MainAxisAlignment.center,
                children: [
                  const Text('Don\'t have an account?'),
                  TextButton(
                    onPressed: () => state.changeStep(
                      AuthenticatorStep.signUp,
                    ),
                    child: const Text('Sign Up'),
                  ),
                ],
              ),
            );
          case AuthenticatorStep.signUp:
            return CustomScaffold(
              state: state,
              body: SignUpForm(),
              footer: Row(
                mainAxisAlignment: MainAxisAlignment.center,
                children: [
                  const Text('Already have an account?'),
                  TextButton(
                    onPressed: () => state.changeStep(
                      AuthenticatorStep.signIn,
                    ),
                    child: const Text('Sign In'),
                  ),
                ],
              ),
            );
          case AuthenticatorStep.confirmSignUp:
            return CustomScaffold(
              state: state,
              body: ConfirmSignUpForm(),
              footer: null,
            );
          default:
            return null;
        }
      },
      child: MultiBlocProvider(
        providers: [
          // ommitted for brevity
        ],
          child: MaterialApp(
            title: 'Shimmers',
            theme: ThemeData(
              primaryColor: Colors.deepPurple,
              brightness: Brightness.dark,
              textTheme: TextTheme(
                headline1: TextStyle(fontSize: 36, fontWeight: FontWeight.bold),
                headline2: TextStyle(fontSize: 24, fontWeight: FontWeight.bold),
                bodyText1: TextStyle(fontSize: 18),
              ),
              inputDecorationTheme: InputDecorationTheme(
                border: OutlineInputBorder(),
                contentPadding: EdgeInsets.all(12),
              ),
            ),
            builder: Authenticator.builder(),
            navigatorObservers: [routeObserver],
            home: BlocListener<AppBloc, AppState>(
              listener: (context, state) {
                if (state is GameResetState) {
                  Future.delayed(const Duration(milliseconds: 1000), () {
                    context.read<GameStateBloc>().add(ResetGameState());
                  });
                }
              },
              child: SelectionScreen(),
            ),
          ),
      ),
    );
  }
}
```

**Error Log**

```
======== Exception caught by rendering library =====================================================
The following assertion was thrown during layout:
A RenderFlex overflowed by 18 pixels on the right.

The relevant error-causing widget was: 
  SignInForm SignInForm:file:///Users/jochemkleine/StudioProjects/guessing_game_alpha/lib/main.dart:78:21
The overflowing RenderFlex has an orientation of Axis.horizontal.
The edge of the RenderFlex that is overflowing has been marked in the rendering with a yellow and black striped pattern. This is usually caused by the contents being too big for the RenderFlex.
```


### Screenshots

![WhatsApp Image 2023-09-05 at 11 32 40](https://github.com/aws-amplify/amplify-flutter/assets/10331148/3a224b62-21b0-4c0e-8b05-2dab5ae6be15)


### Platforms

- [ ] iOS
- [X] Android
- [ ] Web
- [ ] macOS
- [ ] Windows
- [ ] Linux

### Flutter Version

3.10.5

### Amplify Flutter Version

1.4.0.

### Deployment Method

Amplify CLI

### Schema

_No response_

**Comment by dnys1** [comment]
Hi @jochemkleine, thanks for bringing this to our attention. I've opened https://github.com/aws-amplify/amplify-flutter/pull/3692 to fix this.

**Comment by jochemkleine** [comment]
Thanks for picking this up @dnys1 !

**Comment by Jordan-Nelson** [comment]
The fix for this was included in v1.4.2. Please let us know if you face any other issues.
